index.js

```diff
@@ -525,9 +525,26 @@ Router.prototype.find = function find (method, path, derivedConstraints) {
         if (matchedParameter === null) return null
         paramEndIndex = pathIndex + matchedParameter[1].length
       } else {
+        const childrenStartCodes = []
+        for (let i = 0; i < node.numberOfChildren; i++) {
+          childrenStartCodes.push(node.childrenKeys[i].charCodeAt(0))
+        }
```

Comment:
Could this be pre-computed and attached to the node?

Reply:
I did it. I factorize the code  and fixed the reset method